tests/manage/monitoring/conftest.py

```diff
@@ -64,51 +73,102 @@ def prometheus_log(info, alert_list):
+        )
+        if not measure_after:
+            start_time = time.time()
 
-    logging_thread = threading.Thread(
-        target=prometheus_log,
-        args=(info, alert_list)
-    )
-    logging_thread.start()
-
-    result = operation()
-    if measure_after:
-        start_time = time.time()
-    passed_time = time.time() - start_time
-    if minimal_time:
-        additional_time = minimal_time - passed_time
-        if additional_time > 0:
-            time.sleep(additional_time)
-    stop_time = time.time()
-    info['run'] = False
-    logging_thread.join()
-    logger.info(f"Alerts found during measurement: {alert_list}")
-    return {
-        'start': start_time,
-        'stop': stop_time,
-        'result': result,
-        'metadata': metadata,
-        'prometheus_alerts': alert_list
-    }
-
-
-@pytest.fixture(scope="session")
-def workload_stop_ceph_mgr():
+        # init logging thread that checks for Prometheus alerts
+        # while workload is running
+        # based on https://docs.python.org/3/howto/logging-cookbook.html#logging-from-multiple-threads
+        info = {'run': True}
+        alert_list = []
+
+        logging_thread = threading.Thread(
+            target=prometheus_log,
+            args=(info, alert_list)
+        )
+        logging_thread.start()
+
+        result = operation()
+        if measure_after:
+            start_time = time.time()
+        passed_time = time.time() - start_time
+        if minimal_time:
+            additional_time = minimal_time - passed_time
+            if additional_time > 0:
+                time.sleep(additional_time)
+        stop_time = time.time()
+        info['run'] = False
+        logging_thread.join()
+        logger.info(f"Alerts found during measurement: {alert_list}")
+        results = {
+            'start': start_time,
+            'stop': stop_time,
+            'result': result,
+            'metadata': metadata,
+            'prometheus_alerts': alert_list
+        }
+        with open(result_file, 'w') as outfile:
+            logger.info(f"Dumping results of measurement into {result_file}")
+            json.dump(results, outfile)
+    return results
+
+
+@pytest.fixture
+def measurement_dir(tmp_path):
+    """
+    Returns directory path where should be stored all results related
+    to measurement. If 'measurement_dir' is provided by config then use it,
+    otherwise new directory is generated.
+
+    Returns:
+        str: Path to measurement directory
+    """
+    if config.ENV_DATA.get('measurement_dir'):
+        measurement_dir = config.ENV_DATA.get('measurement_dir')
+        logger.info(
+            f"Using measurement dir from configuration: {measurement_dir}"
+        )
+    else:
+        measurement_dir = os.path.join(
+            os.path.dirname(tmp_path),
+            'measurement_results'
+        )
+        if not os.path.exists(measurement_dir):
+            os.mkdir(measurement_dir)
```

Comment:
Could this actually happen? Wouldn't it be more useful in the 1st branch of this if block (eg. when one specifies the measurement dir, but it doesn't exist)?

Reply:
That makes sense. I made the part with creating directory if it doesn't exist accessible to both branches.